### PR TITLE
Updates on ProtonTrack object definition

### PR DESCRIPTION
**Pull request by forthommel**
As discussed at [the last RECO/AT meeting](https://indico.cern.ch/event/784929/) the `reco::ProtonTrack` object is set not to derivate anymore from a `reco::Track` (nor a `reco::XXXCandidate`).
Furthermore, merged `reco::ProtonTrackExtra` down on `reco::ProtonTrack` object (and updated the producer algorithms accordingly).

### CODE REVIEW

File: DataFormats/ProtonReco/interface/ProtonTrack.h [hunk 1]
```diff
@@ -66,20 +128,37 @@ namespace reco
...
+      /// uncertainty on reconstructed time at forward detectors
+      float t_err_;
+      /// fractional momentum loss (positive for diffractive protons)
+      float xi_;
+      /// 5x5 covariance matrix
+      CovarianceMatrix covariance_;
+      /// chi-squared
+      float chi2_;
+      /// number of degrees of freedom
+      unsigned int ndof_;
```
Comment (forthommel) [inline DataFormats/ProtonReco/interface/ProtonTrack.h]: Actually, not sure if a `double`/`float` (as used in `reco::Track`) would be more convenient?

File: RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc [hunk 2]
```diff
@@ -354,27 +351,26 @@ void ProtonReconstructionAlgorithm::reconstructFromSingleRP(
...
     const double p = lhcInfo.energy() * (1. - xi);
-    const reco::Track::Vector momentum(0., p * th_y, sign_z * p * cos_th);
+    const reco::ProtonTrack::Vector momentum(0., p * th_y, sign_z * p * cos_th);
 
-    reco::TrackBase::CovarianceMatrix cm;
+    reco::ProtonTrack::CovarianceMatrix cm;
     cm((int)PT::Index::xi, (int)PT::Index::xi) = xi_unc * xi_unc;
     cm((int)PT::Index::th_y, (int)PT::Index::th_y) = th_y_unc * th_y_unc;
 
     reco::ProtonTrack pt(0., 0, vertex, momentum, xi, cm);
```
Comment (forthommel) [inline RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc]: TODO: move it to the `reco::ProtonTrack` constructor?

Comment (forthommel) [reply RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc]: Woops, wrong line (3 below)

File: DataFormats/ProtonReco/interface/ProtonTrack.h (hunk 1 above)
Comment (jan-kaspar) [reply DataFormats/ProtonReco/interface/ProtonTrack.h]: I can't see how ``ndof`` could become non-integer - any idea?

File: RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc (hunk 2 above)
Comment (jan-kaspar) [reply RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc]: I am confused - move what?

File: DataFormats/ProtonReco/interface/ProtonTrack.h (hunk 1 above)
Comment (forthommel) [reply DataFormats/ProtonReco/interface/ProtonTrack.h]: Me neither... but looking at https://github.com/cms-sw/cmssw/blob/master/DataFormats/Candidate/interface/VertexCompositeCandidate.h#L66, https://github.com/cms-sw/cmssw/blob/master/DataFormats/TrackReco/interface/TrackBase.h#L456, ... I am also confused of this "common usage".

File: RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc (hunk 2 above)
Comment (jan-kaspar) [reply RecoCTPPS/ProtonReconstruction/src/ProtonReconstructionAlgorithm.cc]: Yes, this makes sense.

File: DataFormats/ProtonReco/interface/ProtonTrack.h (hunk 1 above)
Comment (jan-kaspar) [reply DataFormats/ProtonReco/interface/ProtonTrack.h]: Personally, I'd go with with integer.

Comment (slava77) [reply DataFormats/ProtonReco/interface/ProtonTrack.h]: the common use was unsigned value a while back, but then it was changed to double/float in 
https://github.com/cms-sw/cmssw/commit/0a7b1e5136be78b5e0806083b7326f7514447fda
This was before I started following the software developments
@llista may know

**Pull request merged**